### DISCUSSION TITLE
What does mask upload do?

### DISCUSSION THREAD

**Discussion by willneeee** [Mikubill/sd-webui-controlnet#2793] [Q&A] [answered]
I've tried checking the documentation, as well as what the webmasters have shared. It seems to me that this feature regards stronger pose control and then style blending with reference images.
I don't know if this is the right way to interpret it.

**Comment by huchenlei** [comment, marked as answer]
It was requested in https://github.com/Mikubill/sd-webui-controlnet/issues/1921. It is essentially a way for you to upload a predrawn mask, instead of drawing it on input image canvas. The mask is currently only used for ControlNet inpaint and IPAdapters (as CLIP mask to ignore part of the image)

**Reply by willneeee** [reply]
thinks!
